src/particle_filter.h

```diff
@@ -62,26 +60,54 @@ class ParticleFilter
 
     ~ParticleFilter();
 
-    void initUniform(const geo::Vec2& min, const geo::Vec2& max, double t_step,
-                     double a_min, double a_max, double a_step);
+    void configure(tue::Configuration config);
+
+    void initUniform(const geo::Vec2& min, const geo::Vec2& max, double a_min, double a_max);
+
+    void resample(std::function<geo::Transform2()> gen_random_pose_function);
 
-    void resample(unsigned int num_samples = 0);
+    unsigned int resampleLimit(unsigned int number_bins);
 
     std::vector<Sample>& samples() { return samples_[i_current_]; }
 
     const std::vector<Sample>& samples() const { return samples_[i_current_]; }
 
     const Sample& bestSample() const;
 
+    const std::vector<Cluster>& clusters() const;
+
     geo::Transform2 calculateMeanPose() const;
 
-    void normalize();
+    void normalize(bool update_filter=false);
 
 private:
 
-    int i_current_;
+    unsigned int min_samples_, max_samples_;
```

Comment:
Would it make sense to separate a number of these members into different classes or to put them in structs?

Reply:
I think these should be part of the PF, as they are configuration of the PF.

Reply:
Just like the plugin, I don't mind this longer list of member variables.